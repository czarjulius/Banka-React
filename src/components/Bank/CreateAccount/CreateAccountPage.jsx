import React from 'react';
import { connect } from 'react-redux';
import { toastr } from 'react-redux-toastr';
import Dashboard from '../../Dashboard/DashboardPage';
import './createAccount.scss';

import { openAccount } from '../../../actions/openAccountAction';

import Loader from '../../Loader/index';

export class CreateAccountPage extends React.Component {
  state = {
    form: {
      type: '',
      address: '',
      nextOfKin: '',
    },
  }
  componentDidMount = () => {
    const token = localStorage.getItem('token');
    if (!token) {
      this.props.history.push('/login', {});
    }
  };

onInputChange = ({ target: { name, value } }) => {
  this.setState(prevState => ({
    ...prevState,
    form: {
      ...prevState.form,
      [name]: value,
    },
  }));
};

submitHandler = () => {
  const { form } = this.state;
  const account = { ...form };
  const { newAccount } = this.props;

  newAccount(account).then((status) => {
    if (status === 201) {
      toastr.success('Account Created Successfully');
      this.props.history.push('/profile');
    } else {
      toastr.error('Oops something went wrong');
      return false;
    }
  });
};

  render() {

    const { errors, isLoading } = this.props;
    if (isLoading) {
      return (
        <div>
          <Loader />
        </div>
      );
    }
    return (
  <Dashboard header_title="Open New Account">
    <div className="dasboard-body">
      <div className="details-container">
        <div className="transactions">
          <form>
            <div className="form-header">
              <p id="registerMessage" >{errors && errors.message}</p>
              <p id="registerSuccess" />
            </div>
            <div className="input">
              <select name="type" type="text" className="form-control accountInput" onChange={this.onInputChange} >
                <option value="">Select type</option>
                <option value="current">Current</option>
                <option value="savings">Savings</option>
              </select>
              <input type="text" placeholder="address" name="address" className="accountInput" onChange={this.onInputChange} />
              <input type="text" placeholder="Next of kin mobile number" className="accountInput"  name="nextOfKin" onChange={this.onInputChange} />
            </div>
            <p className="sign-up">
              <input type="button" value="Create" className="btn" onClick={this.submitHandler} />
            </p>
          </form>
        </div>
      </div>
    </div>
  </Dashboard>
);
}
}

export const mapStateToProps = state => ({
  isLoading: state.accountReducer.isLoading,
  errors: state.accountReducer.errors,
  accounts: state.accountReducer,
});
export const mapDispatchToProps = dispatch => ({
  newAccount: data => dispatch(openAccount(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CreateAccountPage);
