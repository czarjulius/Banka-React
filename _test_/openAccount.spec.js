import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { CreateAccountPage } from '../src/components/Bank/CreateAccount/CreateAccountPage';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const renderAccountsPage = args => {
    const defaultProps = {
        isLoading: false,
        history: {push: jest.fn()}, 
        match: {},
        accounts: {},
        newAccount: jest.fn(() => Promise.resolve('this is an account...')),
        errors: undefined,
        location: {}
      };
  const props = { ...defaultProps, ...args }; 
  
  return shallow (
    <BrowserRouter>
      <CreateAccountPage {...props} />
    </BrowserRouter>
  );
};
const state = {
  form: {
    type: 'jjj',
    address: 'jdj',
    nextOfKin: 'ffj',
  },
}
describe('Accounts Components Tests', () => {
  it('renders Accounts template', () => {
    const wrapper = renderAccountsPage();
    const wrapperDive = wrapper.dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapperDive.find('.dasboard-body').length).toBe(1);
  });
  it('Find div on the template', () => {
    const wrapper = renderAccountsPage();
    const wrapperDive = wrapper.dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapperDive.find('div').length).toBe(5);
  });
  it('Find select on the template', () => {
    const wrapper = renderAccountsPage();
    const wrapperDive = wrapper.dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapperDive.find('select').length).toBe(1);
  });
  it('Find form on the template', () => {
    const wrapper = renderAccountsPage();
    const wrapperDive = wrapper.dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapperDive.find('form').length).toBe(1);
  });
  it('Find p on the template', () => {
    const wrapper = renderAccountsPage();
    const wrapperDive = wrapper.dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapperDive.find('p').length).toBe(3);
  });
  it('Find option on the template', () => {
    const wrapper = renderAccountsPage();
    const wrapperDive = wrapper.dive().dive().dive();
    expect(wrapper).toMatchSnapshot();
    expect(wrapperDive.find('option').length).toBe(3);
    const newWrapper = wrapperDive.instance()
    newWrapper.state = state;
    newWrapper.submitHandler();
  });

  
});
