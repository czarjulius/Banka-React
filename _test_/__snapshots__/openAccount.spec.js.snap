// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts Components Tests Find div on the template 1`] = `ShallowWrapper {}`;

exports[`Accounts Components Tests Find form on the template 1`] = `ShallowWrapper {}`;

exports[`Accounts Components Tests Find option on the template 1`] = `ShallowWrapper {}`;

exports[`Accounts Components Tests Find p on the template 1`] = `ShallowWrapper {}`;

exports[`Accounts Components Tests Find select on the template 1`] = `ShallowWrapper {}`;

exports[`Accounts Components Tests renders Accounts template 1`] = `ShallowWrapper {}`;
